### PR TITLE
major rewrite of how to deploy a new hub

### PR DESCRIPTION
**Pull request by shaneknapp**
comments more than welcome. 

all changes tested locally via rst2html.py

### CODE REVIEW

**Comment by shaneknapp** [comment]
TODO before merging:  add section about hacking `node-placeholder/settings.yaml`

File: docs/admins/howto/new-hub.rst [hunk 1]
```diff
@@ -25,139 +25,196 @@ The major reasons for making a new hub are:
...
+====================
+
+Name the hub
+------------
+Choose the ``<hubname>`` (typically the course or department). This is permanent.
+
+Determine deployment needs
+--------------------------
+Before creating a new hub, have a discussion with the instructor about the system requirements,
+frequency of assignments and how much storage will be required for the course. Typically, there
```
Comment (ryanlovett) [inline docs/admins/howto/new-hub.rst:51]: Maybe solicit app/tool info too, like whether the class will use Python, R, or Julia; or RStudio, Shiny, linux desktop; what critical libraries, etc.

File: docs/admins/howto/new-hub.rst [hunk 2]
```diff
@@ -25,139 +25,196 @@ The major reasons for making a new hub are:
 
 If your reason is something else, it probably needs some justification :)
 
+Prereqs
+=======
+Working installs of the following utilities:
+  - sops
+  - hubploy
+  - gcloud/kubectl
+  - `cookiecutter <https://github.com/audreyr/cookiecutter>`_
```
Comment (ryanlovett) [inline docs/admins/howto/new-hub.rst:34]: Is it worthwhile to add info about authorization, i.e. what roles the admin user needs to deploy using the above commands.

**Review by ryanlovett — APPROVED** [review]
Looks good! Just a couple of minor suggestions.

**Comment by shaneknapp** [comment]
> TODO before merging: add section about hacking `node-placeholder/settings.yaml`

done w/last commit

File: docs/admins/howto/new-hub.rst (hunk 1 above)
Comment (shaneknapp) [reply docs/admins/howto/new-hub.rst:51]: see https://github.com/berkeley-dsep-infra/datahub/pull/4094/commits/90151f26f32061ecb5dad4c53b2519377b44fd9b

File: docs/admins/howto/new-hub.rst (hunk 2 above)
Comment (shaneknapp) [reply docs/admins/howto/new-hub.rst:34]: see https://github.com/berkeley-dsep-infra/datahub/pull/4094/commits/1d76d09f1852cd718bd84c9fdc557f8cc45c5277

Comment (shaneknapp) [reply docs/admins/howto/new-hub.rst:34]: lemme know if i've missed anything...

**Comment by shaneknapp** [comment]
@ryanlovett is this missing anything?  i was curious about running `hubploy` manually (like i have been) but it appears that CircleCI will take care of that for us.  pls to be confirming y/n.  :)

**Comment by ryanlovett** [comment]
I think it looks good, though I didn't rebuild the docs and view locally, so the git diff formatting might be hiding something from my eyes. Anything run on CI can be run manually too. I tend to do the latter when I need rapid turnaround, like if I'm not 100% sure that something will work. I use the CI config to tell me what to do and when if I haven't deployed in a while.

**Comment by shaneknapp** [comment]
> I think it looks good, though I didn't rebuild the docs and view locally, so the git diff formatting might be hiding something from my eyes. 

i just ran `conda install docutils`, which gives you `rst2html.py` , which gives you a TON of useful stuff!

```/home/sknapp/miniconda3/envs/gcpdev/bin/rst2html.py   /home/sknapp/miniconda3/envs/gcpdev/bin/rst2odt_prepstyles.py
/home/sknapp/miniconda3/envs/gcpdev/bin/rst2html4.py  /home/sknapp/miniconda3/envs/gcpdev/bin/rst2pseudoxml.py
/home/sknapp/miniconda3/envs/gcpdev/bin/rst2html5.py  /home/sknapp/miniconda3/envs/gcpdev/bin/rst2s5.py
/home/sknapp/miniconda3/envs/gcpdev/bin/rst2latex.py  /home/sknapp/miniconda3/envs/gcpdev/bin/rst2xetex.py
/home/sknapp/miniconda3/envs/gcpdev/bin/rst2man.py    /home/sknapp/miniconda3/envs/gcpdev/bin/rst2xml.py
/home/sknapp/miniconda3/envs/gcpdev/bin/rst2odt.py    /home/sknapp/miniconda3/envs/gcpdev/bin/rstpep2html.py
```

i just ran the edited rst file through the html renderer, worked through the formatting errors and then saved it to html and opened it in chrome.  it doesn't have the datahub theme, but you can confirm that the formatting works as intended.

```
(gcpdev) ➜  howto git:(update-new-hub-docs) ✗ rst2html.py new-hub.rst > HUB.html
new-hub.rst:36: (WARNING/2) Block quote ends without a blank line; unexpected unindent.
(gcpdev) ➜  howto git:(update-new-hub-docs) ✗ rst2html.py new-hub.rst > HUB.html
```

and voila!
image

> Anything run on CI can be run manually too. I tend to do the latter when I need rapid turnaround, like if I'm not 100% sure that something will work. I use the CI config to tell me what to do and when if I haven't deployed in a while.

word.  i'll leave things as-is...

**Pull request merged**